.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch: {}
+
+jobs:
+  test-release:
+    name: test-release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '20'
+
+      - name: Install PNPM
+        shell: bash
+        run: |
+          npm i -g pnpm
+          pnpm --version
+
+      - name: Dependency Cache
+        id: dependency-cache
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: pnpm-20-${{ runner.os }}-${{ hashFiles('**/pnpm-lock.yaml') }}
+
+      - name: Install Dependencies
+        if: ${{ steps.dependency-cache.outputs.cache-hit != 'true' }}
+        shell: bash
+        run: pnpm i
+
+      - name: Release new versions
+        shell: bash
+        run: |
+          git checkout ${{ github.head_ref }}
+          git config --local user.name "${{ vars.GH_APP__KONG__BOT_USERNAME }}"
+          git config --local user.email "${{ vars.GH_APP__KONG__BOT_EMAIL }}"
+          pnpm version:dry-run
```

Comment:
This workflow must be changed to have a separate job for each package to  trigger based on `paths-changed-filter` GH action for each package and run the npm build step to ensure no node build errors rather than just installing dependencies and doing nothing. 

Ex: https://github.com/Kong/shared-actions/blob/main/.github/workflows/ci.yaml#L108-L173

Reply:
To limit the size of this CI build file over time as more independent job-per-package are added, you can use a pattern to use `workflow_call` for each package to build in a separate yml (reusable workflow) file and then use the workflow file within this build job. Helps with manageability of file size and keeps things decoupled better.

Ex: 
- Parent build workflow for mesh: https://github.com/Kong/kong-mesh/blob/master/.github/workflows/kuma-build-test-distribute.yaml
- Invokes reusable workflows to test: https://github.com/Kong/kong-mesh/blob/master/.github/workflows/kuma-build-test-distribute.yaml#L126 managed within a separate file https://github.com/Kong/kong-mesh/blob/master/.github/workflows/kuma-_test.yaml

Reply:
This is missing the `commitlint` to check the enforced commit structure before geenerating the changelog
https://github.com/Kong/shared-actions/blob/main/.github/workflows/ci.yaml#L43-L47

Reply:
changed updated